tests/common/plugins/sanity_check/__init__.py

```diff
@@ -377,9 +377,9 @@ def recover_on_sanity_check_failure(ptfhost, duthosts, failed_results, fanouthos
                         recover_method)
 
     except BaseException as e:
-        request.config.cache.set(cache_key, True)
-        add_custom_msg(request, f"{DUT_CHECK_NAMESPACE}.{cache_key}", True)
-        add_custom_msg(request, f"{DUT_CHECK_NAMESPACE}.{recovery_cache_key}", False)
+        request.config.cache.set(sanity_failed_cache_key, True)
+        add_custom_msg(request, f"{DUT_CHECK_NAMESPACE}.{sanity_failed_cache_key}", True)
```

Comment:
sanity_failed_cache_key shouldn't be updated here,
It should only present for the first sanity check fail or not,
logically, it shouldn't be included in the logic of sanity recover

Reply:
But please make sure why we set a cache here and won't affect this logic